packages/flutter_tools/lib/src/reporting/usage.dart

```diff
@@ -467,3 +465,154 @@ class LogToFileAnalytics extends AnalyticsMock {
+
+  @override
+  bool enabled = true;
+
+  @override
+  bool suppressAnalytics = false;
+
+  @override
+  String get clientId => 'test-client';
+
+  @override
+  Future<void> ensureAnalyticsSent() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Stream<Map<String, dynamic>> get onSend => throw UnimplementedError();
+
+  @override
+  void printWelcome() { }
+
+  @override
+  void sendCommand(String command, {Map<String, String> parameters}) {
+    commands.add(TestUsageCommand(command, parameters: parameters));
+  }
+
+  @override
+  void sendEvent(String category, String parameter, {String label, int value, Map<String, String> parameters}) {
+    events.add(TestUsageEvent(category, parameter, label: label, value: value, parameters: parameters));
+  }
+
+  @override
+  void sendException(dynamic exception) {
+    exceptions.add(exception);
+  }
+
+  @override
+  void sendTiming(String category, String variableName, Duration duration, {String label}) {
+    timings.add(TestTimingEvent(category, variableName, duration, label: label));
+  }
+}
+
+@immutable
+class TestUsageCommand {
+  const TestUsageCommand(this.command, {this.parameters});
+
+  final String command;
+  final Map<String, String> parameters;
+
+  @override
+  bool operator ==(Object other) {
+    return other is TestUsageCommand &&
+      other.command == command &&
+      _mapsEqual(other.parameters, parameters);
+  }
+
+  @override
+  int get hashCode => command.hashCode ^ parameters.hashCode;
+
+  @override
+  String toString() => 'TestUsageCommand($command, parameters:$parameters)';
+}
+
+@immutable
+class TestUsageEvent {
+  const TestUsageEvent(this.category, this.parameter, {this.label, this.value, this.parameters});
+
+  final String category;
+  final String parameter;
+  final String label;
+  final int value;
+  final Map<String, String> parameters;
+
+  @override
+  bool operator ==(Object other) {
+    return other is TestUsageEvent &&
+      other.category == category &&
+      other.parameter == parameter &&
+      other.label == label &&
+      other.value == value &&
+      _mapsEqual(other.parameters, parameters);
+  }
+
+  @override
+  int get hashCode => category.hashCode ^
+    parameter.hashCode ^
+    label.hashCode ^
+    value.hashCode ^
+    parameters.hashCode;
+
+  @override
+  String toString() => 'TestUsageEvent($category, $parameter, label:$label, value:$value, parameters:$parameters)';
+}
+
+@immutable
+class TestTimingEvent {
+  const TestTimingEvent(this.category, this.variableName, this.duration, {this.label});
+
+  final String category;
+  final String variableName;
+  final Duration duration;
+  final String label;
+
+  @override
+  bool operator ==(Object other) {
+    return other is TestTimingEvent &&
+      other.category == category &&
+      other.variableName == variableName &&
+      other.duration == duration &&
+      other.label == label;
+  }
+
+  @override
+  int get hashCode => category.hashCode ^
```

Comment:
These aren't good hashcodes and I don't even care, it does not matter for these objects